id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/helper/IdRepoServiceHelper.java

```diff
@@ -0,0 +1,165 @@
+        if (Objects.isNull(schemaVersion) || schemaVersion.contentEquals("null")) {
+            mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER, "getSchema",
+                    "\n" + "schemaVersion is null");
+            throw new IdRepoAppUncheckedException(MISSING_INPUT_PARAMETER.getErrorCode(),
+                    String.format(MISSING_INPUT_PARAMETER.getErrorMessage(),
+                            ROOT_PATH + IdObjectValidatorConstant.PATH_SEPERATOR + "IDSchemaVersion"));
+        }
+        try {
+            if (schemaMap.containsKey(schemaVersion)) {
+                return schemaMap.get(schemaVersion);
+            } else {
+                RestRequestDTO restRequest;
+                restRequest = restBuilder.buildRequest(RestServicesConstants.SYNCDATA_SERVICE, null, ResponseWrapper.class);
+                restRequest.setUri(restRequest.getUri().concat("?schemaVersion=" + schemaVersion));
+                ResponseWrapper<Map<String, String>> response = restHelper.requestSync(restRequest);
+                schemaMap.put(schemaVersion, response.getResponse().get("schemaJson"));
+                supportedHandlesInSchema.put(schemaVersion, getSupportedHandles(response.getResponse().get("schemaJson")));
+                return getSchema(schemaVersion);
+            }
+        } catch (IdRepoDataValidationException | RestServiceException e) {
+            mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER, "getSchema", "\n" + e.getMessage());
+            throw new IdRepoAppUncheckedException(IdRepoErrorConstants.SCHEMA_RETRIEVE_ERROR);
+        }
+    }
+
+    public Map<String, HandleDto> getSelectedHandles(Object identity) throws IdRepoAppException {
+        Map<String, Object> requestMap = convertToMap(identity);
+        if (requestMap.containsKey(ROOT_PATH) && Objects.nonNull(requestMap.get(ROOT_PATH))) {
+            Map<String, Object> identityMap = (Map<String, Object>) requestMap.get(ROOT_PATH);
+            String schemaVersion = String.valueOf(identityMap.get(ID_SCHEMA_VERSION));
+            if(identityMap.containsKey(SELECTED_HANDLES) && Objects.nonNull(identityMap.get(SELECTED_HANDLES))) {
+                mosipLogger.debug(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER, "getSelectedHandles",
+                        requestMap.get(SELECTED_HANDLES));
+                List<String> selectedHandleFieldIds = (List<String>) identityMap.get(SELECTED_HANDLES);
+                return selectedHandleFieldIds.stream()
+                        .filter( handleFieldId -> supportedHandlesInSchema.get(schemaVersion).contains(handleFieldId))
+                        .collect(Collectors.toMap(handleName->handleName,
+                                handleFieldId-> {
+                                    String handle = ((String) identityMap.get(handleFieldId))
+                                            .concat("@").concat(handleFieldId)
```

Comment:
We should have a map between fieldId and the actual handle type that can be used. Please create a JIRA task

Reply:
https://mosip.atlassian.net/browse/ES-572